### PR TITLE
update interface proto annotations

### LINKED ISSUE #1155: clean up `{accept,implement}_interface` in protos
we need to keep in sync with using full-scoped interface names in protos (already merged in cosmos-sdk)

- [reference PR](https://github.com/cosmos/cosmos-sdk/pull/14476)
- [issue describing problem](https://github.com/cosmos/cosmos-sdk/issues/14353)
- [new updated docs with required changes](https://github.com/cosmos/cosmos-sdk/blob/a0da913e86e43fd8cb1eafb49f5473e976d0b514/UPGRADING.md#acceptsimplements_interface-proto-annotations)

I can make a PR, making the issue for reference

cc @amaurym

### PR DESCRIPTION
**Pull request by pyramation**
closes #1155

### CODE REVIEW

File: proto/cosmwasm/wasm/v1/authz.proto [hunk 1]
```diff
@@ -85,14 +85,14 @@ message CombinedLimit {
 // message.
 // Since: wasmd 0.30
 message AllowAllMessagesFilter {
-  option (cosmos_proto.implements_interface) = "ContractAuthzFilterX";
+  option (cosmos_proto.implements_interface) = "cosmwasm.wasm.ContractAuthzFilterX";
```
Comment (pyramation) [inline proto/cosmwasm/wasm/v1/authz.proto:88]: 1. [registered interface here](https://github.com/cosmology-tech/wasmd/blob/5c39daee99b838482ec1bdd980bfa0a56213ec34/x/wasm/types/codec.go#L79)
2. [interface defined here](https://github.com/cosmology-tech/wasmd/blob/5c39daee99b838482ec1bdd980bfa0a56213ec34/x/wasm/types/authz.go#L216) 

would this make it `cosmwasm.wasm.ContractAuthzFilterX`?

File: proto/cosmwasm/wasm/v1/authz.proto [hunk 2]
```diff
@@ -57,7 +57,7 @@ message MaxCallsLimit {
 // MaxFundsLimit defines the maximal amounts that can be sent to the contract.
 // Since: wasmd 0.30
 message MaxFundsLimit {
-  option (cosmos_proto.implements_interface) = "ContractAuthzLimitX";
+  option (cosmos_proto.implements_interface) = "cosmwasm.wasm.ContractAuthzLimitX";
```
Comment (pyramation) [inline proto/cosmwasm/wasm/v1/authz.proto:60]: 1. [registered interface here](https://github.com/cosmology-tech/wasmd/blob/5c39daee99b838482ec1bdd980bfa0a56213ec34/x/wasm/types/codec.go#L87)
2. [interface defined here](https://github.com/cosmology-tech/wasmd/blob/5c39daee99b838482ec1bdd980bfa0a56213ec34/x/wasm/types/authz.go#L198)

would this make it `cosmwasm.wasm.ContractAuthzLimitX`?

File: proto/cosmwasm/wasm/v1/types.proto [hunk 3]
```diff
@@ -90,7 +90,7 @@ message ContractInfo {
   // Extension is an extension point to store custom metadata within the
   // persistence model.
   google.protobuf.Any extension = 7
-      [ (cosmos_proto.accepts_interface) = "ContractInfoExtension" ];
+      [ (cosmos_proto.accepts_interface) = "cosmwasm.wasm.ContractInfoExtension" ];
```
Comment (pyramation) [inline proto/cosmwasm/wasm/v1/types.proto:93]: 1. [registered interface here](https://github.com/cosmology-tech/wasmd/blob/5c39daee99b838482ec1bdd980bfa0a56213ec34/x/wasm/types/codec.go#L77)
2. [interface defined here](https://github.com/cosmology-tech/wasmd/blob/5c39daee99b838482ec1bdd980bfa0a56213ec34/x/wasm/types/types.go#L187)

**Review by alpe — COMMENTED** [review]
Thanks a lot for this PR. Too bad that I did all of this in #1149 already for sdk 0.47 as well as adding the signer annotation. Merging your PR will create merge conflicts for the sdk 0.47 branch. 

Are the annotations something you need already in the next wasmd release or can this wait?
if yes, please add the version to the cosmwasm interface names and run `make proto-all`

**Comment by pyramation** [comment]
> Thanks a lot for this PR. Too bad that I did all of this in #1149 already for sdk 0.47 as well as adding the signer annotation. Merging your PR will create merge conflicts for the sdk 0.47 branch.
> 
> Are the annotations something you need already in the next wasmd release or can this wait? if yes, please add the version to the cosmwasm interface names and run `make proto-all`

Ideally we can get these ASAP, I'd be down to help make another PR to avoid the conflicts. Let me know the best approach!

**Comment by alpe** [comment]
Thanks for the updates! 
Please run `make proto-all` to format the proto files and have updated Go files.
When done, I can include this for the v0.31.0 release.

Not sure how to address the merge issues with #1149 afterwards. I may simply revert this PR so let's make sure #1149 contains all the proto options that you need, too.

**Comment by alpe** [comment]
@pyramation please see my last comment, to get this into v0.31.0

**Comment by pyramation** [comment]
> @pyramation please see my last comment, to get this into v0.31.0

should I maybe do this again, but branch off of #1149 ?

**Comment by alpe** [comment]
@pyramation just run `make proto-all` on this branch to have proper protobuf and generated Go files. This is only relevant if you need them for the v0.31 release. If not, let's close this PR without merging

I would like to get #1149 (which includes all the protobuf options of this PR already) into main as soon as the release is cut.

**Comment by alpe** [comment]
fyi: I have tagged a rc0 release. There is still time on Monday to get this into the final one

**Comment by pyramation** [comment]
great! I just ran the `make proto-all` and pushed :)

**Pull request merged**